src/rpcwallet.cpp

```diff
@@ -1963,6 +1963,14 @@ Value walletpassphrase(const Array& params, bool fHelp)
     pwalletMain->TopUpKeyPool();
 
     int64_t nSleepTime = params[1].get_int64();
+    // Timeout limits:
+    // If the timeout is larger than the max value of int64 (922337203685477580), nSleepTime will be negative
+    // If the timeout is larger than 999999000000000, boost will not be able to convert it to seconds
+    if (nSleepTime < 0 || nSleepTime > 999999000000000)
```

Comment:
Can we throw the warning before even converting by just comparing it to the max of int64_t? Also I think the comment of 922337203685477580 is missing the last digit - I think its 9223372036854775807.

`
#include <limits>
if (nSleepTime > std::numeric_limits<int64_t>::max()) { throw JSONRPCError(); }
`

It would be best if we could find the actual type that posix_time::seconds() uses, as it could possibly change based on compiler, architecture, etc. Best I could find was that it was a long from http://www.boost.org/doc/libs/1_55_0/doc/html/date_time/posix_time.html#date_time.posix_time.time_duration but I'm not confident if that's the same function. How did you come up with an approx value of 999999000000000?

Reply:
You are right on both points:
- Finding a compiler-independant fix would be much better
- I forgot the last digit in the comment ^^

I found the value with some tests, and tried to find the exact type but I have not managed yet.
However, I have just had an idea of a way to fix it based on a Boost type.
I will submit a PR update shortly.